ext/strscan/strscan.c

```diff
@@ -710,18 +719,25 @@ strscan_do_scan(VALUE self, VALUE pattern, int succptr, int getstr, int headonly
     }
     else {
         StringValue(pattern);
-        rb_encoding *enc = rb_enc_check(p->str, pattern);
         if (S_RESTLEN(p) < RSTRING_LEN(pattern)) {
+            if (!(strscan_ascii_compat_fastpath(p->str) && strscan_ascii_compat_fastpath(pattern))) {
```

Comment:
Is this equivalent though?
```
> Encoding.compatible? "é".b, "é"
=> nil
```
So these are not compatible but they are both one of the 3 simple encodings

Reply:
Yeah, you're right. `rb_enc_check` end up in `enc_compatible_latter`, and if it returns `0` it raises: https://github.com/ruby/ruby/blob/43b059b6a3b5c49b7d883c49dd1200580c1f92be/encoding.c#L1068-L1128

So I'm missing a check for both encindex to be equal.

Reply:
I replaced it to just a fastpath if encindex are the same, which I think would most often be the case.